sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -296,6 +296,8 @@ case class Generate(
 
   override protected def withNewChildInternal(newChild: LogicalPlan): Generate =
     copy(child = newChild)
+
+  override lazy val references: AttributeSet = generator.references
```

Comment:
The default implementation is kind of `generator.references -- producedAttributes`. Ideally these two should have no conjunction, and is the same as `generator.references`.

Reply:
so if it is valid bug, we should target at fixing `producedAttributes`?